src/org/openstreetmap/josm/actions/ToggleEditLockLayerAction.java

```diff
@@ -0,0 +1,66 @@
+// License: GPL. For details, see LICENSE file.
+package org.openstreetmap.josm.actions;
+
+import static org.openstreetmap.josm.tools.I18n.tr;
+
+import java.awt.Component;
+import java.awt.event.ActionEvent;
+import java.util.List;
+
+import javax.swing.AbstractAction;
+import javax.swing.JCheckBoxMenuItem;
+
+import org.openstreetmap.josm.data.osm.Lockable;
+import org.openstreetmap.josm.gui.dialogs.LayerListDialog;
+import org.openstreetmap.josm.gui.layer.AbstractModifiableLayer;
+import org.openstreetmap.josm.gui.layer.Layer;
+import org.openstreetmap.josm.gui.layer.Layer.LayerAction;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+/**
+ * An action enabling/disabling the {@linkplain AbstractModifiableLayer#lock() read-only flag}
+ * of the layer specified in the constructor.
+ * 
+ * @since XXX
```

Comment:
Hmmm, a comment starting with uppercase `XXX` has the added benefit that it's recognized as a TODO by a lot of editors. Is it okay if I simply modify the regex in the script to also accept uppercase XXX?

Reply:
Maybe, but we are going to want to tag bastiK and stoecker on it -- I don't know if there was a reason to only do lower case `xxx`.

Reply:
Here's the PR for it in any case: https://github.com/JOSM/josm/pull/101

Reply:
Stupid question: Is there any reason why you want the `@since XXX` to be flagged as a TODO? Especially since it should be replaced by a script?

Reply:
No particular reason and it probably doesn't really matter. Might be useful to highlight it in case something gets missed somehow (however unlikely that might be). The main thing is that I don't really see a reason against doing it.

Reply:
Changed the xxx to lowercase